controllers/clusters/helpers.go

```diff
@@ -202,10 +204,37 @@ func createSpecDifferenceMessage(k8sSpec, iSpec any) (string, error) {
 		return "", err
 	}
 
-	msg := "There are external changes on the Instaclustr console. Please reconcile the specification manually. "
-	specDifference := fmt.Sprintf("k8s spec: %s; data from instaclustr: %s", k8sData, iData)
+	var k8sSpecMap map[string]any
+	err = json.Unmarshal(k8sData, &k8sSpecMap)
+	if err != nil {
+		return "", err
+	}
+
+	var iSpecMap map[string]any
+	err = json.Unmarshal(iData, &iSpecMap)
+	if err != nil {
+		return "", err
+	}
+
+	const externalChangesBaseMsg = "There are external changes on the Instaclustr console. Please reconcile the specification manually."
+
+	diffs := dcomparison.MapsDiff("spec", k8sSpecMap, iSpecMap)
```

Comment:
Let's move the `"spec"` parameter to the constants also

Reply:
Done